File: src/main/java/oracle/model/entity/Job.java
package oracle.model.entity;

import org.springframework.lang.Nullable;

import javax.persistence.Column;
import javax.persistence.Entity;

@Entity(name = "jobs")
public class Job {
    @javax.persistence.Id
    @Column(unique = true)
    private Integer jobID;
    private String jobTitle;
    @Nullable
    private Integer minSalary;
    @Nullable
    private Integer maxSalary;

    public Job() {
    }

    public Integer getJobID() {
        return jobID;
    }

    public void setJobID(Integer jobID) {
        this.jobID = jobID;
    }

    public String getJobTitle() {
        return jobTitle;
    }

    public void setJobTitle(String jobTitle) {
        this.jobTitle = jobTitle;
    }

    @Nullable
    public Integer getMinSalary() {
        return minSalary;
    }

    public void setMinSalary(@Nullable Integer minSalary) {
        this.minSalary = minSalary;
    }

    @Nullable
    public Integer getMaxSalary() {
        return maxSalary;
    }

    public void setMaxSalary(@Nullable Integer maxSalary) {
        this.maxSalary = maxSalary;
    }
}

File: frontend/src/app/components/jobs/job.ts

export interface JobInterface {
  jobID: string;
  jobTitle: string;
  minSalary: number;
  maxSalary: number;
}

export class Job {
  constructor(fields: Partial<JobInterface>) {
    this.jobID = fields.jobID;
    this.jobTitle = fields.jobTitle;
    this.minSalary = fields.minSalary;
    this.maxSalary = fields.maxSalary;
  }

  jobID: string;
  jobTitle: string;
  minSalary: number;
  maxSalary: number;



}
